### PR TITLE
feat: Drop support for aiohttp, flask and pyramid. Refactor fastapi/starlette interface.

### PR DESCRIPTION
**Pull request by EdgyEdgemond**
BREAKING CHANGE: Python 3.8 support dropped, aiohttp/flask/pyramid support dropped, generate_handler interface refactored.
Refs: #16

### CODE REVIEW

**Comment by EdgyEdgemond** [comment]
closes #16

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        version: [3.8, 3.9.18, 3.10.13, 3.11, 3.12]
+        version: [3.9.18, 3.10.13, 3.11, 3.12]
```
Comment (JoseKilo) [inline .github/workflows/tests.yml:69]:
```suggestion
        version: [3.9, 3.10, 3.11, 3.12]
```
do you need to specify the patch version for some reason ?

File: web_error/cors.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from __future__ import annotations
+
+import dataclasses
+
+
+@dataclasses.dataclass
+class CorsConfiguration:
+    allow_origins: list[str] = dataclasses.field(default_factory=lambda: ["*"])
+    allow_methods: list[str] = dataclasses.field(default_factory=lambda: ["*"])
+    allow_headers: list[str] = dataclasses.field(default_factory=lambda: ["*"])
```
Comment (JoseKilo) [inline web_error/cors.py:10]: The default values here are very open, maybe I'd let the caller specify them explicitly 

if it makes tests less verbose, we can also define a pytest fixture with the `*` values

File: web_error/error.py [hunk 3]
```diff
@@ -23,12 +23,16 @@ def __init__(
...
-    def marshal(self: typing.Self) -> dict[str, str]:
-        return {
+    def marshal(self: typing.Self, *, strip_debug: bool = False) -> dict[str, str]:
+        ret = {
             "code": self.code,
             "message": self.message,
             "debug_message": self.debug_message,
         }
+        if strip_debug:
+            ret.pop("debug_message")
```
Comment (JoseKilo) [inline web_error/error.py:33]: Minor:

```suggestion
            del ret["debug_message"]
```

the key is always present at this point

... or even
```python
if not strip_debug:
    ret["debug_message"] = self.debug_message
```

**Review by JoseKilo — APPROVED** [review]
LGTM :rocket:

File: .github/workflows/tests.yml (hunk 1 above)
Comment (EdgyEdgemond) [reply .github/workflows/tests.yml:69]: yeah, it blows up installing 3.10 etc and claims only `3.10.13` available, pinning shuts it up

File: web_error/cors.py (hunk 2 above)
Comment (EdgyEdgemond) [reply web_error/cors.py:10]: :+1:

File: web_error/error.py (hunk 3 above)
Comment (EdgyEdgemond) [reply web_error/error.py:33]: We strip out more in the next PR, otherwise yes :)

**Pull request merged**